packages/frontend/app/routes/tenants.create.tsx

```diff
@@ -0,0 +1,152 @@
+import { createTenantSchema } from '~/lib/validate.server'
+import type { ZodFieldErrors } from '~/shared/types'
+import { checkAuthAndRedirect } from '../lib/kratos_checks.server'
+import { type LoaderFunctionArgs } from '@remix-run/node'
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const cookies = request.headers.get('cookie')
+  await checkAuthAndRedirect(request.url, cookies)
+  return null
+}
+
+export default function CreateTenantPage() {
+  const response = useActionData<typeof action>()
+  const { state } = useNavigation()
+  const isSubmitting = state === 'submitting'
+
+  return (
+    <div className='pt-4 flex flex-col space-y-4'>
+      <div className='flex flex-col rounded-md bg-offwhite px-6'>
+        <PageHeader>
+          <h3 className='text-xl'>Create Tenant</h3>
+          <Button aria-label='go back to tenants page' to='/tenants'>
+            Go to tenants page
+          </Button>
+        </PageHeader>
+        {/* Create Tenant form */}
+        <Form method='post' replace>
+          <div className='px-6 pt-5'>
+            <ErrorPanel errors={response?.errors.message} />
+          </div>
+
+          <fieldset disabled={isSubmitting}>
+            {/* Tenant General Info */}
+            <div className='grid grid-cols-1 px-0 py-3 gap-6 md:grid-cols-3 border-b border-pearl'>
+              <div className='col-span-1 pt-3'>
+                <h3 className='text-lg font-medium'>General Information</h3>
+              </div>
+              <div className='md:col-span-2 bg-white rounded-md shadow-md'>
+                <div className='w-full p-4 space-y-3'>
+                  <Input
+                    name='publicName'
+                    label='Public Name'
+                    placeholder='Public name'
+                    error={response?.errors?.fieldErrors.publicName}
+                  />
+                  <Input
+                    name='email'
+                    label='Email'
+                    placeholder='Email'
+                    error={response?.errors?.fieldErrors.email}
+                  />
+                </div>
+              </div>
+            </div>
+            {/* Tenant General Info - END */}
+            {/* Tenant Sensitive Info */}
+            <div className='grid grid-cols-1 px-0 py-3 gap-6 md:grid-cols-3 border-b border-pearl'>
+              <div className='col-span-1 pt-3'>
+                <h3 className='text-lg font-medium'>Sensitive Information</h3>
+              </div>
+              <div className='md:col-span-2 bg-white rounded-md shadow-md'>
+                <div className='w-full p-4 space-y-3'>
+                  <PasswordInput
+                    name='apiSecret'
+                    label='API Secret'
+                    placeholder='The API secret. Treat as sensitive information.'
+                    error={response?.errors?.fieldErrors.apiSecret}
+                    required
+                  />
```

Comment:
I think we should probably update the api secret that is set on the cookie when we save this. Otherwise we will update the api secret and continue using the old one. Although I think there is a bug with updating the api secret atm.

Reply:
@BlairCurrey applied the new apiSecret from the front end, but as you mentioned, the resolver still needs attention.

https://github.com/interledger/rafiki/pull/3254/commits/ae7692531e043fa080810102665c2f616fcd1b6b